docs/reference/mapping/runtime.asciidoc

```diff
@@ -0,0 +1,518 @@
+[[runtime]]
+== Runtime fields
+Typically, you index data into {es} to promote faster search. However, indexing
+can be slow and requires more disk space, and you have to reindex your data to
+add fields to existing documents. With _runtime fields_, you can add
+fields to documents already indexed to {es} without reindexing your data.
+
+[discrete]
+[[runtime-benefits]]
+=== Benefits
+Because runtime fields aren't indexed, you can more quickly ingest raw data
```

Comment:
The key benefit of runtime fields is the ability to add fields to documents after these documents were already ingested into Elasticsearch. In addition to that, adding a runtime field does not increase the index size and runtime fields do not require indexing, so the ingest process performs better. 

A user can define a runtime field initially, and after working with it for a while can decide to change that runtime field into an indexed field in the template, to obtain better query performance. The result will be that this field will be indexed for any new document to any index that will be generated from this template. Queries that will refer to the runtime field will not need to change to refer to the indexed field, and in fact can refer to some indices in which the field is a runtime field and to other indices in which the field is an indexed field.